trustfall_core/test_data/tests/valid_queries/fold_with_no_outputs_transform_and_filter_greater_than.graphql.ron

```diff
@@ -0,0 +1,22 @@
+TestGraphQLQuery (
+    // This @fold statement doesn't actually need to fully evaluate:
+    // it has no outputs, and the predicate can be determined to be satisfied
+    // with only a partial evaluation of the @fold.
```

Comment:
I don't think this is true currently, actually. In order to be `> 2`, the fold needs to see three elements -- and there are only three prime factors, so this `@fold` essentially gets fully evaluated. (Yes, small technicality here, Trustfall doesn't have to _determine_ that is has fully evaluated it by verifying that it has drained its iterator, but it will have drained it all the same.)

Consider making the argument be `1` instead of `2`?

Reply:
Yeah I can change it, but I actually wrote this comment after testing with my patch, it does result in a diff after my patch, so it was partially skipped. I'll change it though, no problem.

Reply:
With `> 1` as the predicate I think this is fine.

I suspect the change you were previously observing was just that `InputIteratorExhausted` trace operations were being elided, that's the technicality I was referring to. But I like this more explicit version better.